ci/pipelines/toolbox-staging-ecr.yml

```diff
@@ -0,0 +1,160 @@
+    type: git
+    icon: github
+    source:
+      uri: https://github.com/alphagov/pay-ci
+      branch: master 
+      paths:
+        - ci/pipelines/toolbox-test-ecr.yml
+  # ECR Test registry resource
+  - name: ecr-registry-test
+    type: dev-registry-image
+    icon: docker
+    source:
+      repository: govukpay/toolbox
+      aws_access_key_id: ((readonly_access_key_id))
+      aws_secret_access_key: ((readonly_secret_access_key))
+      aws_session_token: ((readonly_session_token))
+      aws_role_arn: arn:aws:iam::((pay_aws_test_account_id)):role/concourse
+      aws_ecr_registry_id: "((pay_aws_test_account_id))"
+      aws_region: eu-west-1
+      tag: ready-staging
+  # ECR Staging registry resource
+  - name: ecr-registry-staging
+    type: dev-registry-image
+    icon: docker
+    source:
+      repository: govukpay/toolbox
+      aws_access_key_id: ((readonly_access_key_id))
+      aws_secret_access_key: ((readonly_secret_access_key))
+      aws_session_token: ((readonly_session_token))
+      aws_role_arn: arn:aws:iam::((pay_aws_staging_account_id)):role/concourse
+      aws_ecr_registry_id: "((pay_aws_staging_account_id))"
+      aws_region: eu-west-1
+
+resource_types:
+  # Custom resource type - this has been merged to the
+  # https://github.com/concourse/registry-image-resource repo,
+  # however is not yet in a stable release. Using this custom
+  # resource type allows us to support the ECR registry_id parameter
+  # This can be removed once registry_id is in stable release of 'registry-image'.
+  - name: dev-registry-image
+    type: registry-image
+    source:
+      repository: concourse/registry-image-resource
+      username: ((docker-username))
+      password: ((docker-password))
+      tag: dev
+
+jobs:
+  - name: update-toolbox-staging-ecr-pipeline
+    plan:
+      - get: toolbox-staging-ecr
+        trigger: true
+      - set_pipeline: toolbox-staging-ecr
+        file: toolbox-staging-ecr/ci/pipelines/toolbox-staging-ecr.yml
+  - name: toolbox-staging-approval
+    plan:
+      - get: ecr-registry-test
+        trigger: true
+        params:
+          format: oci
+      - task: pull-tags-from-ecr-registry-test
+        config:
+          platform: linux
+          image_resource:
+            type: registry-image
+            source:
+              repository: amazon/aws-cli
+          params:
+            AWS_ACCESS_KEY_ID: ((readonly_access_key_id))
+            AWS_SECRET_ACCESS_KEY: ((readonly_secret_access_key))
+            AWS_SESSION_TOKEN: ((readonly_session_token))
+            AWS_ASSUME_ROLE_ARN: arn:aws:iam::((pay_aws_test_account_id)):role/concourse
+            AWS_REGISTRY_ID: "((pay_aws_test_account_id))"
+            AWS_REGION: eu-west-1
+            AWS_PAGER: ""
+            AWS_DEFAULT_OUTPUT: text
+          inputs:
+            - name: ecr-registry-test
+          outputs:
+            - name: docker_tags
+          run:
+            path: /bin/sh
+            args:
```

Comment:
I think it's worth adding a comment here explaining why we're having to do the `assume-role` manually, instead of using another `dev-registry-image` resource. Something like:

```
# Use the AWS CLI to list images in the test account ECR. This command is not available in the `dev-registry-
# image` resource, so we also need to manually assume the role to gain access.
```

Reply:
I think I've added something suitable.